orangecontrib/text/nyt.py

```diff
@@ -47,25 +29,23 @@ def parse_record_json(record, includes_metadata):
                 else:
                     if doc[field]:
                         field_value = doc[field]
-                string_document += field_value
                 metas_row.append(field_value)
         # Add the pub_date.
-        field_value = ""
-        if "pub_date" in doc and doc["pub_date"]:
-            field_value = doc["pub_date"]
-        metas_row.append(field_value)
+        metas_row.append(doc.get("pub_date", ""))
         # Add the glocation.
         metas_row.append(",".join([kw["value"] for kw in doc["keywords"] if kw["name"] == "glocations"]))
 
         # Add the section_name.
-        class_val = ""
-        if "section_name" in doc and doc["section_name"]:
-            class_val = doc["section_name"]
+        class_val = doc.get("section_name", "")
 
-        documents.append(string_document)
+        if class_val is None:
```

Comment:
Instead of filling the missing values with empty strings I believe it might be better to actually insert those values as missing since Orange supports missing values, don't you? For example, you can look at the adult.tab dataset that comes along with Orange. I think all you need to do is to set the corresponding entires in the numpy array to nan. Please, fix this for class as well as for meta attributes (county codes, etc.).